.github/workflows/action-build.yml

```diff
@@ -15,13 +19,24 @@ on:
       slug:
         required: true
         type: string
+      arch:
+        required: false
+        type: string
+      sign:
+        default: false
+        type: boolean
 
 jobs:
-  build:
+  build-sign-upload:
+    permissions:
+      id-token: write
     name: build for ${{ inputs.slug }}
     runs-on: ${{ inputs.os }}
     steps:
-      - uses: actions/checkout@v3
+      - name: describe runner
+        run: |
+          echo "Running job with os: '${{ inputs.os }}', arch: '${{ inputs.arch }}', slug: '${{ inputs.slug }}', runtime: '${{ inputs.runtime }}', target: '${{ inputs.target }}'"
```

Comment:
Maybe you can write this such that it shows up as an annotation? https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-setting-a-notice-message

Reply:
I am not sure if I understand the benefits of creating annotations. The example showed in the doc has a file name and a line number, which isn't applied to what we want to show here. 

Perhaps, a job summary would be better suited? GITHUB_STEP_SUMMARY

Reply:
I've added `error` and `notice` annotations to logs.